models/value_set.go

```diff
@@ -56,13 +56,13 @@ func (v ValueSetMap) CodeDisplayWithPreferredCodeForField(entry *Entry, coded *C
 		log.Fatalln(err)
 	}
 	codeDisplay.MapDataCriteria = MapDataCriteria
-	for FieldOidInd := 0; FieldOidInd < len(MapDataCriteria.FieldOids[field]); FieldOidInd++ {
-		codeDisplay.PreferredCode = coded.PreferredCode(codeDisplay.PreferredCodeSets, codeDisplay.CodeSetRequired, codeDisplay.ValueSetPreferred, v, MapDataCriteria.FieldOids[field][FieldOidInd])
+	for i := range MapDataCriteria.FieldOids[field] {
```

Comment:
you want `for i, oid`. `i` will be the index and `oid` will be the value at that index in each iteration.